src/main/java/com/gtnewhorizons/gtnhgradle/modules/ToolchainModule.java

```diff
@@ -167,31 +172,30 @@ public void apply(GTNHGradlePlugin.@NotNull GTNHExtension gtnh, @NotNull Project
                     }
 
                     jc.setSourceCompatibility(String.valueOf(javaVersion));
-                    jc.getOptions()
-                        .getRelease()
-                        .set(8);
+                    // For JABEL mode: set --release 8 to restrict to J8 APIs
+                    // For JVMDG/MODERN: leave release unset (null) to allow modern stdlib APIs
+                    if (mode == ModernJavaSyntaxMode.JABEL) {
+                        jc.getOptions()
+                            .getRelease()
+                            .set(8);
+                    } else if (mode.requiresModernStdlib()) {
+                        jc.getOptions()
+                            .getRelease()
+                            .set((Integer) null);
+                    }
                     jc.getJavaCompiler()
                         .set(compiler);
                 });
 
             if (useJabel) {
                 deps.add(JavaPlugin.ANNOTATION_PROCESSOR_CONFIGURATION_NAME, UpdateableConstants.NEWEST_JABEL);
+                deps.add("testAnnotationProcessor", UpdateableConstants.NEWEST_JABEL);
```

Comment:
Use the named constant `JavaPlugin.TEST_ANNOTATION_PROCESSOR_CONFIGURATION_NAME` here